source/fuzz/fuzzer.cpp

```diff
@@ -228,8 +228,11 @@ Fuzzer::Fuzzer(std::unique_ptr<opt::IRContext> ir_context,
   MaybeAddFinalPass<FuzzerPassAdjustMemoryOperandsMasks>(&final_passes_);
   MaybeAddFinalPass<FuzzerPassAdjustSelectionControls>(&final_passes_);
   MaybeAddFinalPass<FuzzerPassAddNoContractionDecorations>(&final_passes_);
-  MaybeAddFinalPass<FuzzerPassInterchangeSignednessOfIntegerOperands>(
-      &final_passes_);
+  if (!fuzzer_context_->IsWgslCompatible()) {
+    // Signedness is not as interchangeable in WGSL as in SPIR-V.
```

Comment:
The SPIR-V reader in Tint is supposed to handle signedness mismatches between operands of an instruction.

For instructions like OpIAdd, it bitcasts the operands and result type to be the same as the signedness of the first operand.
For instructions where the signedness of the semantics is determined by the opcode, e.g. OpSLessThan, then operands are coerced to the expected signedness.  If the WGSL output is wrong, then you may be seeing a Tint spirv-reader bug.
 It bitcasts operands and return types to one to be the same as the other (e.g. for OpIAdd)

Reply:
Thanks for the comment. I disabled this since I had lots of `Binary expression has invalid operands` errors and I thought that this transformation invalidates WGSL spec. I'll add an issue for this to the tint repo.

Reply:
Added an issue https://crbug.com/tint/666.